apollo-federation/src/correctness/query_plan_analysis.rs

```diff
@@ -356,12 +356,15 @@ fn interpret_fetch_node(
     conditions: &[Literal],
     fetch: &FetchNode,
 ) -> Result<ResponseShape, String> {
-    let mut result = if let Some(_requires) = &fetch.requires {
+    let mut result = if !fetch.requires.is_empty() {
         // TODO: check requires
-        compute_response_shape_for_entity_fetch_operation(&fetch.operation_document, schema)
-            .map(|rs| rs.add_boolean_conditions(conditions))
+        compute_response_shape_for_entity_fetch_operation(
+            fetch.operation_document.as_parsed().unwrap(),
```

Comment:
Is it better to `unwrap()` here or could we meaningfully propagate an error (`String`) ?